### PR TITLE
Simplify some bits around rayquery handling

### PR DESCRIPTION
**Pull request by nhaehnle**
The two commits are self-contained, but the point here is to remove the `hasRayQuery` virtual method which wasn't implemented consistently.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit 8a1ba393776cfa7a7c697a0d8aa0eb39e6770338
<details><summary>CTS tests (Failed: 0/213064)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 40075/71024 (56.4%)</li>
<li>Failed: 0/71024 (0.0%)</li>
<li>Not Supported: 30949/71024 (43.6%)</li><li>Warnings: 0/71024 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 40113/71014 (56.5%)</li>
<li>Failed: 0/71014 (0.0%)</li>
<li>Not Supported: 30901/71014 (43.5%)</li><li>Warnings: 0/71014 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 8.6, Gfx10</summary><ul>

<li>Passed: 36219/71026 (51.0%)</li>
<li>Failed: 0/71026 (0.0%)</li>
<li>Not Supported: 34807/71026 (49.0%)</li><li>Warnings: 0/71026 (0.0%)</li>
</ul></details>


</ul></details>

File: imported/llvm-dialects
Comment (Flakebi) [inline imported/llvm-dialects:1]: Unintended change?

**Review by Flakebi — COMMENTED** [review]
LGTM, but there seems to be a stray submodule update

**Comment by jiaolu** [comment]
LGTM

**Review by amdrexu — APPROVED** [review]

**Review by amdrexu — APPROVED** [review]

File: imported/llvm-dialects
Comment (nhaehnle) [reply imported/llvm-dialects:1]: Thanks for spotting that, that snuck in accidentally.

**Comment by nhaehnle** [comment]
Removed the unintended change to the llvm-dialects submodule pointer

**Comment by amdvlk-admin** [comment]
### Test summary for commit 4d896f14c958dde3a149c87098e454bd6cf5cec7
<details><summary>CTS tests (Failed: 0/213061)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 40074/71023 (56.4%)</li>
<li>Failed: 0/71023 (0.0%)</li>
<li>Not Supported: 30949/71023 (43.6%)</li><li>Warnings: 0/71023 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 40112/71013 (56.5%)</li>
<li>Failed: 0/71013 (0.0%)</li>
<li>Not Supported: 30901/71013 (43.5%)</li><li>Warnings: 0/71013 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 8.6, Gfx10</summary><ul>

<li>Passed: 36218/71025 (51.0%)</li>
<li>Failed: 0/71025 (0.0%)</li>
<li>Not Supported: 34807/71025 (49.0%)</li><li>Warnings: 0/71025 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by nhaehnle** [comment]
ping

**Review by amdrexu — APPROVED** [review]

**Comment by nhaehnle** [comment]
Merged as part of #2468

**Pull request closed without merging**